### ISSUE TITLE
Specifying name_prefix in aws_cloudwatch_event_rule causes resource to be created twice

### ISSUE THREAD

**Issue by otterley** [terraform-providers/terraform-provider-aws#4547]
### Summary

Specifying `name_prefix` instead of `name` as a parameter to an `aws_cloudwatch_event_rule` causes the provider to create the resource twice.  One of them, however, will be orphaned by Terraform and will not be tracked in the state, leaving the Rule configured with CloudWatch even after a destroy operation until manually deleted by the user.

### Terraform Version

Terraform v0.11.7
+ provider.aws v1.18.0

### Affected Resource(s)

* aws_cloudwatch_event_rule

### Terraform Configuration Files

```hcl
resource "aws_cloudwatch_event_rule" "ebs_backup" {
  name_prefix         = "ebs-backup-"
  # note: using "name" instead cures the problem
  description         = "Back up ${var.volume_name} every ${var.frequency}"
  schedule_expression = "rate(${var.frequency})"
  is_enabled          = "${var.enable_event_rule}"
}
```

### Expected Behavior

A single CloudWatch Event Rule should be created with the specified name prefix.

### Actual Behavior

Two CloudWatch Event Rules are created with the specified name prefix.

### Steps to Reproduce

Declare a resource similar to the one above and apply it.

**Comment by bflad** [comment]
Thanks for reporting this. Bug fix pull request submitted: #4579

**Comment by bflad** [comment]
The fix for this will release with v1.20.0 of the AWS provider, likely later today. 👍

**Comment by bflad** [comment]
This has been released in [version 1.20.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1200-may-23-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
